docker/docker_image_dest.go

```diff
@@ -120,15 +145,26 @@ func (d *dockerImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobI
 			return types.BlobInfo{Digest: inputInfo.Digest, Size: size}, nil
 		}
 	}
+	mountQuery := prepareMountQuery(d.mount, inputInfo.Digest.String())
```

Comment:
`inputInfo.Digest` may be unknown (`""`) at this time (e.g. when the blob is being compressed).  In that case the mounting should probably not be attempted. (The `docker/distribution` implementation ignores the `mount` parameter if it is empty, but that is not a documented part of the API AFAICS.)